gotham/src/middleware/session/backend/mod.rs

```diff
@@ -27,17 +31,26 @@ pub trait Backend: Send {
     /// Persists a session, either creating a new session or updating an existing session.
     fn persist_session(
         &self,
+        state: &State,
         identifier: SessionIdentifier,
         content: &[u8],
-    ) -> Result<(), SessionError>;
+    ) -> Pin<Box<SetSessionFuture>>;
```

Comment:
Sorry for taking so long to review this - I believe this type signature is incorrect? Without any lifetime information, the `GetSessionFuture` type needs to live for `'static`, which it will not do if it accessed the state in an async context. I think this needs to be changed to something like

```rust
persist_session<'fut, 'a: 'fut, 'b: 'fut, 'c: 'fut>(
    &'a self,
    state: &'b State,
    identifier: SessionIdentifier,
    content: &'c [u8]
) -> Pin<Box<GetSessionFuture + 'fut>>
```

Alternatively, it might make sense to let `#[async_trait]` deal with the lifetimes.

Reply:
No worries, thanks for reviewing. I'll try to get to these shortly.

The last time I used `#[async_trait]` it resulting in the rust compiler not being able to produce any decent error messages for any errors in the code that implemented those traits. We ended up changing back to returning Futures instead. Do you know if this problem has been resolved? I would be hesitant to use it unless it has..

Reply:
The only time I used `#[async_trait]` was to run `cargo expand` on it to see how it worked because I didn't want my proc macro to invoke another proc macro. So I don't know if this has been changed or has ever been a problem. However, I feel like the code would probably be more readable if we don't introduce this many lifetimes, but if you run into any incorrect/unhelpful error messages, just write the code the explicit way. I'm happy with both solutions.